pkg/didmethod/peer/resolver.go

```diff
@@ -21,7 +22,7 @@ func NewDIDResolver(store *DIDStore) *DIDResolver {
 }
 
 // Read implements didresolver.DidMethod.Read interface (https://w3c-ccg.github.io/did-resolution/#resolving-input)
-func (resl *DIDResolver) Read(did string, versionID interface{}, versionTime string, noCache bool) ([]byte, error) {
+func (resl *DIDResolver) Read(did string, opts ...didresolver.Options) ([]byte, error) {
```

Comment:
I thought the purpose of the refactor was to able to return a doc. I still see []byte.

Reply:
I do like moving to an ...Option idiom rather than a long list or optional arguments in the signature. Later in the PR i am confused about using a struct rather than functional arguments,